templates/masterclasses.html

```diff
@@ -1,70 +1,119 @@
+{%- endif -%}
+{%- if slot == 'image' -%}
+<div class="p-image-container--3-2 u-hide--medium u-hide--small">
+  {{ image(url="https://assets.ubuntu.com/v1/3a4d4155-Training.svg",
+    alt="",
+    width="480",
+    height="240",
+    hi_def=True,
+    loading="auto",
+    attrs={"class": "p-image-container__image"}
+    ) | safe
+  }}
+</div>
+{%- endif -%}
+{% endcall -%}
 
-      <a href="/videos" class="p-button--positive">Explore Catalog</a>
-      <a class="p-button" id="lucky-button">I'm feeling lucky</a>
 
-    </div>
-    <div class="col u-align--center u-hide--medium u-hide--small u-vertically-center">
-      <img src="https://res.cloudinary.com/canonical/image/fetch/f_auto,q_auto,fl_sanitize,h_366/https://assets.ubuntu.com/v1/3a4d4155-Training.svg"
-        srcset="https://res.cloudinary.com/canonical/image/fetch/f_auto,q_auto,fl_sanitize,w_900,h_732/https://assets.ubuntu.com/v1/3a4d4155-Training.svg 2x"
-        alt="" width="450" height="366" loading="auto">
-    </div>
-  </div>
-</section>
+{# prepare list of up to 4 upcoming sessions #}
+{# TODO: move the logic to backend? #}
+{% set upcoming = [] %}
+{% for video in live_videos %}
+  {% if upcoming|length < 4 %}
+    {% set _=upcoming.append([video, 'live']) %}
+  {% endif %}
+{% endfor %}
 
+{% if upcoming_videos_24h and shown|length < 4 %}
+  {% set _=upcoming.append([upcoming_videos_24h[0], 'soon']) %}
+{% endif %}
 
-{% if live_videos or upcoming_videos_24h or upcoming_videos_future %}
-<section class="">
-  <div class="u-fixed-width">
-    <hr>
-  </div>
-  <div class="row">
-    <div class="col-8">
-      <h4>Upcoming sessions</h4>
-      <ul class="p-list">
-        {% set shown = [] %}
-        {% for video in live_videos %}
-          {% if shown|length < 2 %}
-            {{ upcoming_session(video, 'live', now) }}
-            {% set _=shown.append(1) %}
-          {% endif %}
-        {% endfor %}
+{% if upcoming_videos_future and shown|length < 4 %}
+  {% set _=upcoming.append([upcoming_videos_future[0], 'future']) %}
+{% endif %}
 
-        {% if upcoming_videos_24h and shown|length < 2 %}
-          {{ upcoming_session(upcoming_videos_24h[0], 'soon', now) }}
-          {% set _=shown.append(1) %}
-        {% endif %}
+{%- call(slot) vf_tiered_list(is_list_full_width_on_tablet=true) -%}
+{%- if slot == 'title' -%}
+<h2>Upcoming sessions</h2>
+{%- endif -%}
 
-        {% if upcoming_videos_future and shown|length < 2 %}
-          {{ upcoming_session(upcoming_videos_future[0], 'future', now) }}
-          {% set _=shown.append(1) %}
-        {% endif %}
-      </ul>
-    </div>
-  </div>
-</section>
-{% endif %}
+{%- for session in upcoming -%}
+{%- set video = session[0] -%}
+{%- set session_type = session[1] -%}
+
+{%- if slot == ('list_item_title_' ~ loop.index) -%}
+{%- if session_type == 'live' -%}
+  <a href="{{ video.stream }}">
+    <h3 class="p-heading--5">
+      <span style="margin-right: 0.3rem;" class="live-dot"></span>
```

Comment:
```suggestion
      <span style="position: relative;top: -.1rem;left: .3rem;" class="live-dot"></span>
```
Maybe?

Reply:
Yes, that could work as a quick hack. I haven't touched those styles yet, but we can improve on that later if needed.